examples/seismic/tutorials/04_dask.ipynb

```diff
@@ -178,41 +180,40 @@
     "        'observed' data using the real model. For a real case\n",
     "        the worker would be reading seismic data from disk.\n",
     "    \"\"\"\n",
-    "    true_model = get_true_model()\n",
-    "    shot_id = param['shot_id']\n",
-    "    \n",
-    "    src_coordinates = np.empty((1, len(param['shape'])))\n",
-    "    src_coordinates[0, :] = [30, param['shot_id']*1000./(param['nshots']-1)]\n",
+    "    # Reconstruct objects\n",
+    "    with open(\"arguments.pkl\", \"rb\") as cp_file:\n",
+    "        cp = pickle.load(cp_file)\n",
+    "        \n",
+    "    solver = cp['solver']\n",
+    "    geometry = cp['geometry']\n",
+    "    # source position changes according to the index\n",
+    "    shot_id=param['shot_id']\n",
     "    \n",
-    "    # Number of receiver locations per shot.\n",
-    "    nreceivers = 101\n",
-    "\n",
-    "    # Set up receiver data and geometry.\n",
-    "    rec_coordinates = np.empty((nreceivers, len(param['shape'])))\n",
-    "    rec_coordinates[:, 1] = np.linspace(0, true_model.domain_size[0], num=nreceivers)\n",
-    "    rec_coordinates[:, 0] = 980. # 20m from the right end\n",
+    "    geometry.src_positions[0,:]=[30, shot_id*1000./(param['nshots']-1)]\n",
```

Comment:
what is the point of saving the Geometry if modified right after reading it?

Reply:
Sure, you are right, there is no need to save the `geometry`. In this example, the geometry is the same, the only thing that changes is the source position. In this case, I could change the `dump_shot_data` function to save only the source positions instead of `geometry` and just to assign the `src_positions` to the `solver.geometry` in the `fwi_gradient_i` function.